dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/teradata/TeradataLogsJdbcTask.java

```diff
@@ -0,0 +1,80 @@
+package com.google.edwmigration.dumper.application.dumper.connector.teradata;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.common.io.ByteSink;
+import com.google.edwmigration.dumper.application.dumper.connector.SqlQueryFactory;
+import com.google.edwmigration.dumper.application.dumper.connector.teradata.TeradataLogsConnector.SharedState;
+import com.google.edwmigration.dumper.application.dumper.handle.JdbcHandle;
+import com.google.edwmigration.dumper.application.dumper.task.AbstractJdbcTask;
+import com.google.edwmigration.dumper.application.dumper.task.TaskRunContext;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.function.BiFunction;
+import java.util.function.BiPredicate;
+import javax.annotation.Nonnull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.dao.DataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.ResultSetExtractor;
+
+public class TeradataLogsJdbcTask extends AbstractJdbcTask<Void> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(TeradataLogsConnector.class);
+
+  @VisibleForTesting public static String EXPRESSION_VALIDITY_QUERY = "SELECT TOP 1 %s FROM %s";
+
+  private final SharedState state;
+  private final SqlQueryFactory factory;
+
+  /* pp */ TeradataLogsJdbcTask(
+      @Nonnull String targetPath, SharedState state, SqlQueryFactory factory) {
+    super(targetPath);
+    this.state = Preconditions.checkNotNull(state, "SharedState was null.");
```

Comment:
Parameters are by default not null. I would suggest to remove the `checkNotNull` call (similarly as you already don't check whether `factory` parameter is not null).